files/default/sql/ddl/updates/1.0.0.sql

```diff
@@ -13,4 +13,287 @@ CREATE TABLE IF NOT EXISTS `secrets` (
+                                                FOREIGN KEY (`on_demand_feature_group_id`) REFERENCES
+                                               `hopsworks`.`on_demand_feature_group`(`id`)
+                                               ON DELETE CASCADE
+                                               ON UPDATE NO ACTION;
+
+ALTER TABLE `hopsworks`.`feature_group` ADD COLUMN `feature_group_type` INT(11) NOT NULL DEFAULT '0';
+
+ALTER TABLE `hopsworks`.`feature_group` ADD COLUMN `on_demand_feature_group_id` INT(11) NULL;
+ALTER TABLE `hopsworks`.`feature_group` ADD CONSTRAINT `on_demand_feature_group_fk`
+                                               FOREIGN KEY (`on_demand_feature_group_id`) REFERENCES
+                                               `hopsworks`.`on_demand_feature_group`(`id`)
+                                               ON DELETE CASCADE
+                                               ON UPDATE NO ACTION;
+
+ALTER TABLE `hopsworks`.`feature_group` ADD COLUMN `cached_feature_group_id` INT(11) NULL;
+ALTER TABLE `hopsworks`.`feature_group` ADD CONSTRAINT `cached_feature_group_fk`
+                                                FOREIGN KEY (`cached_feature_group_id`) REFERENCES
+                                               `hopsworks`.`cached_feature_group`(`id`)
+                                               ON DELETE CASCADE
+                                               ON UPDATE NO ACTION;
+
+ALTER TABLE `hopsworks`.`training_dataset` ADD  COLUMN `training_dataset_type`   INT(11) NOT NULL DEFAULT '0';
+
+ALTER TABLE `hopsworks`.`training_dataset` ADD COLUMN `hopsfs_training_dataset_id` INT(11) NULL;
+ALTER TABLE `hopsworks`.`training_dataset` ADD CONSTRAINT `hopsfs_training_dataset_fk`
+                                                FOREIGN KEY (`hopsfs_training_dataset_id`) REFERENCES
+                                               `hopsworks`.`hopsfs_training_dataset`(`id`)
+                                               ON DELETE CASCADE
+                                               ON UPDATE NO ACTION;
+
+ALTER TABLE `hopsworks`.`training_dataset` ADD COLUMN `external_training_dataset_id` INT(11) NULL;
+ALTER TABLE `hopsworks`.`training_dataset` ADD CONSTRAINT `external_training_dataset_fk`
+                                                FOREIGN KEY (`external_training_dataset_id`) REFERENCES
+                                               `hopsworks`.`external_training_dataset`(`id`)
+                                               ON DELETE CASCADE
+                                               ON UPDATE NO ACTION;
+
+
+/*
+  Move columns from feature_group to cached_feature_group
+*/
+
+-- Move hive_tbl_id
+INSERT INTO `hopsworks`.`cached_feature_group`(`offline_feature_group`)
+SELECT `hive_tbl_id` FROM `hopsworks`.`feature_group` WHERE feature_group_type=0;
+
+-- Set foreign key on parent table
+UPDATE `hopsworks`.`feature_group` INNER JOIN `hopsworks`.`cached_feature_group`
```

Comment:
I don't this will work on flyway. If I remember correctly you are not allowed to have an update statement without `WHERE` condition

Reply:
Does flyway only support a subset of the operations? is there an easy way for me to test flyway? because it does work when using the mysql client manually.

The way I have tested this migration manually is as follows:

1. Spin up master hopsworks VM
2. Create a bunch of feature groups and training datasets using Feature Store V1
3. Apply all SQL updates manually in order
4. Deploy Feature Store V2 .ear and .war
5. Verify that all featurestore metdata is there and was migrated successfully.

Reply:
@SirOibaf do you mean flyway sets the `SQL_SAFE_UPDATES` to `ON`?

Reply:
@tkakantousis @SirOibaf I've added it now